### PR TITLE
Allowing overflow set to be reversed

### PR DESCRIPTION
**Pull request by mltejera**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000 // Should I file an issue simply for tracking purposes?
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Our scenario requires reversed visual behavior (rtl) but we'd still like to maintain proper keyboard navigation (ltr). We can achieve this simply by reversing the DOM order of the elements.

#### Focus areas to test

Make sure elements behave and reflow as you'd expect.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/12115)

### CODE REVIEW

**Comment by mltejera** [comment]
@brandonthomas FYI

File: packages/office-ui-fabric-react/src/components/OverflowSet/OverflowSet.base.tsx [hunk 1]
```diff
@@ -9,6 +9,8 @@ import { IOverflowSet, IOverflowSetItemProps, IOverflowSetProps, IOverflowSetSty
 
 const getClassNames = classNamesFunction<IOverflowSetStyleProps, IOverflowSetStyles>();
 
+const defaultIsReversed = false;
```
Comment (brandonthomas) [inline packages/office-ui-fabric-react/src/components/OverflowSet/OverflowSet.base.tsx]: I think you can just keep this inline with your destructure

Comment (mltejera) [reply packages/office-ui-fabric-react/src/components/OverflowSet/OverflowSet.base.tsx]: I have to destructure twice though, so I'd rather it be tied to a constant.

File: packages/office-ui-fabric-react/src/components/OverflowSet/OverflowSet.types.ts [hunk 2]
```diff
@@ -54,6 +54,13 @@ export interface IOverflowSetProps extends React.ClassAttributes<OverflowSetBase
    */
   vertical?: boolean;
 
+  /**
+   * Controls wether or not the DOM order is reversed.
```
Comment (brandonthomas) [inline packages/office-ui-fabric-react/src/components/OverflowSet/OverflowSet.types.ts]: specify what is reversed. It's not quite the whole thing, just the placement of the overflow button vs the rest.

Comment (mltejera) [reply packages/office-ui-fabric-react/src/components/OverflowSet/OverflowSet.types.ts]: updated to only reverse overflow position

File: packages/office-ui-fabric-react/src/components/OverflowSet/OverflowSet.types.ts [hunk 3]
```diff
@@ -54,6 +54,13 @@ export interface IOverflowSetProps extends React.ClassAttributes<OverflowSetBase
    */
   vertical?: boolean;
 
+  /**
+   * Controls wether or not the overflow button is placed on the right or left of the items.
+   * This gives a reveresed visual behavior but maintains correct keyboard navigation.
+   * @defaultValue false
+   */
+  isOverflowReversed?: boolean;
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/OverflowSet/OverflowSet.types.ts]: now about "overflowSide": "start" | "end"

with default being end

**Comment by mltejera** [comment]
@micahgodbolt I'm getting linter errors from deprecated api's that aren't related to my change.

Are those going to block this getting in?

**Comment by micahgodbolt** [comment]
yeah, linting error is on "toolbar" one of the new controls pulled over from teams. you certainly aren't to blame there. merge master back in and give it another shot.

**Comment by mltejera** [comment]
Looks like the linter errors are specific on over flow set?

**Comment by micahgodbolt** [comment]
Ah that makes sense. FocusZone got moved to it's own package. The old import is deprecated.

**Pull request merged**